clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1638,6 +1638,36 @@ invariant_checker_t::check_regdeps_invariants(per_shard_t *shard, const memref_t
     // Check markers.
     if (memref.marker.type == TRACE_TYPE_MARKER) {
         switch (memref.marker.marker_type) {
+        case TRACE_MARKER_TYPE_FILETYPE:
+            report_if_false(
+                shard,
+                static_cast<offline_file_type_t>(memref.marker.marker_value) ==
+                    shard->file_type_,
+                "TRACE_MARKER_TYPE_FILETYPE and shard file type are not the same");
+            // Check that the file type does not contain any architecture specific
+            // information except OFFLINE_FILE_TYPE_ARCH_REGDEPS.
+            report_if_false(shard,
+                            !TESTANY(OFFLINE_FILE_TYPE_ARCH_AARCH64, shard->file_type_),
+                            "OFFLINE_FILE_TYPE_ARCH_REGDEPS traces cannot contain "
+                            "OFFLINE_FILE_TYPE_ARCH_AARCH64");
+            report_if_false(shard,
+                            !TESTANY(OFFLINE_FILE_TYPE_ARCH_ARM32, shard->file_type_),
```

Comment:
Could combine all of these (TESTANY checks for any of the bits): the message would be less specific but that doesn't seem so bad?

Reply:
Agreed, we don't really need a specific error message for every arch.
Collapsed into one check/error message using `OFFLINE_FILE_TYPE_ARCH_ALL`.
Done.